safenode/src/bin/kadclient.rs

```diff
@@ -149,6 +192,33 @@ async fn get_client_dir() -> Result<PathBuf> {
     Ok(home_dirs)
 }
 
+fn parse_tokens_amount(amount_str: &str) -> Token {
+    use std::str::FromStr;
+    match Token::from_str(amount_str) {
+        Ok(amount) => return amount,
+        Err(err) => match err {
+            sn_dbc::Error::ExcessiveTokenValue => {
+                warn!("Invalid amount to send: {amount_str:?}, it exceeds the maximum possible value.");
+                println!("Invalid amount to send: {amount_str:?}, it exceeds the maximum possible value.");
+            }
+            sn_dbc::Error::LossOfTokenPrecision => {
+                warn!("Invalid amount to send: '{amount_str}', the minimum possible amount is one nano token (0.000000001).");
+                println!("Invalid amount to send: '{amount_str}', the minimum possible amount is one nano token (0.000000001).");
+            }
+            sn_dbc::Error::FailedToParseToken(msg) => {
+                warn!("Invalid amount to send: '{amount_str}': {msg}.");
+                println!("Invalid amount to send: '{amount_str}': {msg}.");
+            }
+            other_err => {
+                warn!("Invalid amount to send: '{amount_str}': {other_err:?}.");
+                println!("Invalid amount to send: '{amount_str}': {other_err:?}.");
+            }
+        },
+    }
+
+    Token::from_nano(0)
```

Comment:
Isn't it perhaps better to just straight error instead of defaulting to 0? I know the reason gets printed, but I'm not sure it's good to randomly default like this.

Reply:
fixed